### ISSUE TITLE
Can I build an index contains millions of concepts search to search a specific short concept in this index?

### ISSUE THREAD

**Issue by liulizuel** [castorini/anserini#1970]
I have a task to recognize a given concept name in the concepts set which consists of all names of many concepts.
I'm wondering that if I have some concepts around several millions, can I take each concept as a document and build concepts index?  In this way, given an input candidate concept, can I quickly search its similar concepts from the index?
Here are some concepts in my collections:
![image](https://user-images.githubusercontent.com/26185859/189026653-45f161e5-0406-45ec-bff5-be089dd5a0a9.png)

**Comment by ABDULAZIZALQATAN** [comment]
Does the concept mean field in your documents? It should be from the same source, right?
As far as I know, you should divide your data collection itself into fields and work on them.

**Comment by lintool** [comment]
Yes, the solution is to use multi-field indexing, and place your concepts in a separate field. For specific examples, look at https://github.com/castorini/anserini#regressions-for-beir-v100 under the "multifield" conditions.

Or this: https://github.com/castorini/pyserini/blob/master/docs/experiments-elastic.md
